### PR TITLE
ENH Avoid potential data disclosure

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Issue
- https://github.com/silverstripe/silverstripe-linkfield/issues/208

### CODE REVIEW

File: client/src/components/LinkField/LinkField.js [hunk 1]
```diff
@@ -353,20 +353,20 @@ const LinkField = ({
...
+        description={linkData.description}
+        versionState={linkData.versionState}
         typeTitle={type.title || ''}
         typeIcon={type.icon}
         onDelete={handleDelete}
         onClick={() => { setEditingID(linkID); }}
         onButtonKeyDownEdit={() => setIsKeyboardEditing(true)}
         onUnpublishedVersionedState={handleUnpublishedVersionedState}
-        canDelete={data[linkID]?.canDelete ? true : false}
+        canDelete={linkData.canDelete ? true : false}
```
Comment (ChloeSartorelli) [inline client/src/components/LinkField/LinkField.js:369]: Unrelated refactoring - this variable is explicitly created above so a) let's use it and b) no need for null safety chaining operator.

File: src/Controllers/LinkFieldController.php [hunk 2]
```diff
@@ -118,11 +118,7 @@ private function getLinkData(Link $link): array
         if (!$link->canView()) {
             $this->jsonError(403);
         }
-        $data = $link->jsonSerialize();
-        $data['canDelete'] = $link->canDelete();
-        $data['description'] = $link->getDescription();
-        $data['versionState'] = $link->getVersionedState();
-        return $data;
+        return $link->getData();
```
Comment (ChloeSartorelli) [inline src/Controllers/LinkFieldController.php:121]: Moved these into the method on `Link` since the only time the method is called is here. Makes it easier to maintain and see the full list of what's passed through to the react component from PHP in one place.

File: src/Models/Link.php [hunk 3]
```diff
@@ -201,33 +201,30 @@ public function onBeforeWrite(): void
...
-
-        $data['Title'] = $this->getTitle();
-
-        return $data;
+        return [
+            'title' => $this->getTitle(),
+            'description' => $this->getDescription(),
+            'canDelete' => $this->canDelete(),
+            'versionState' => $this->getVersionedState(),
+            'typeKey' => $typeKey,
```
Comment (ChloeSartorelli) [inline src/Models/Link.php:219]: Explicitly only these are used in the component.
Even creating super-duper unique link subclasses won't require adding more data here - the modal is handled separately, this is just for display before opening the modal.

There's no way to easily customise the react component, so adding an extension hook to update this data is overkill.

File: src/Models/Link.php [hunk 4]
```diff
@@ -201,33 +201,30 @@ public function onBeforeWrite(): void
...
     }
 
     /**
-     * Return a rendered version of this form.
+     * Return a rendered version of this link.
      *
-     * This is returned when you access a form as $FormObject rather
-     * than <% with FormObject %>
+     * This is returned when you access a link as $LinkRelation or $Me rather
+     * than <% with LinkRelation %>
```
Comment (ChloeSartorelli) [inline src/Models/Link.php:227]: Unrelated refactoring - a link is not a form, this was probably just copy pasted from elsewhere originally.

File: client/src/components/LinkField/LinkField.js [hunk 5]
```diff
@@ -353,20 +353,20 @@ const LinkField = ({
...
       }
-      const type = types.hasOwnProperty(data[linkID]?.typeKey) ? types[data[linkID]?.typeKey] : {};
+      const type = types.hasOwnProperty(linkData.typeKey) ? types[linkData.typeKey] : {};
       links.push(<LinkPickerTitle
         key={linkID}
         id={linkID}
-        title={data[linkID]?.Title}
-        description={data[linkID]?.description}
-        versionState={data[linkID]?.versionState}
+        title={linkData.title}
```
Comment (ChloeSartorelli) [inline client/src/components/LinkField/LinkField.js:360]: Note upper case `Title` is now lower case `title` - all of the data is now using lowerCamelCase key names, as per javascript tradition. It was a mix before.

**Review by emteknetnz — APPROVED** [review]

**Comment by emteknetnz** [comment]
Tested locally, works good
